pkg/sbctl/compatibility.go

```diff
@@ -22,21 +24,47 @@ func init() {
 }
 
 func Decode(resource string, data []byte) (runtime.Object, *schema.GroupVersionKind, error) {
+	originalData := data
 	decode := scheme.Codecs.UniversalDeserializer().Decode
 	decoded, gvk, err := decode(data, nil, nil)
 	if err == nil {
 		return decoded, gvk, nil
 	}
 
-	log.Println("failed to decode file, will try addind list GVK", err)
+	log.Warn("could not to decode data, will try adding list GVK", err)
 	data, err = wrapListData(resource, data)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "failed to wrap data")
+		log.Warn(err)
+	} else {
+		decoded, gvk, err = decode(data, nil, nil)
+		if err != nil {
+			log.Warn("could not decode wrapped data: ", err)
+		}
 	}
 
-	decoded, gvk, err = decode(data, nil, nil)
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "failed to decode wrapped data")
+	if decoded == nil {
+		// Try to decode object into an unstructured object
+		var v unstructured.Unstructured
+		err = json.Unmarshal(originalData, &v)
+		if err == nil {
```

Comment:
value of `err` changes a few times in this block.  how about  logging it every time

```
if err != nil {
   // log and continue
}
```